HW0/P5/P5.txt

```diff
@@ -1 +1,327 @@
-# Your answers here.
+###############################
+
+# CS 205 Fall 2015 Homework 0 Problem 5
+# Submitted by Kendrick Lo (Harvard ID: 70984997)
+# Github username: ppgmg
+
+Summary
+-------
+
+1. 255 seconds
+2. 34 seconds
+3. 8 seconds
+4. log N seconds
+5. 15 seconds
+6. 272 seconds, 143 seconds
```

Comment:
What my understanding for 1 bag is : For total 256 bags, we ignore the handing time first, each person takes 1 bag, so we have 256 groups. Same idea as problem 5, but a little bit different is 256 people should communicate first, it takes an extra 1 second. So the time without handing time is 16 seconds. Now for the handing time, from 256—>1, it takes 8 seconds, but we need 1 extra second to pass bags to cashier. Therefore, the total time will be 16 + 8 + 1 = 25 sec